src/php/ext/grpc/channel_credentials.c

```diff
@@ -236,6 +236,47 @@ PHP_METHOD(ChannelCredentials, createInsecure) {
   RETURN_NULL();
 }
 
+/**
+ * Create XDS channel credentials
+ * @param ChannelCredentials|null $fallback_creds  The fallback credentials used
+ * if the channel target does not have the 'xds:///' scheme or if the xDS
+ * control plane does not provide information on how to fetch credentials
+ * dynamically.
+ * @return ChannelCredentials The xDS channel credentials object
+ */
+PHP_METHOD(ChannelCredentials, createXds) {
+  grpc_channel_credentials* xds_creds = NULL;
+  zval* fallback_creds = NULL;
+  if (zend_parse_parameters_ex(0,  // ZEND_PARSE_PARAMS_QUIET,
+                               ZEND_NUM_ARGS() TSRMLS_CC, "O!", &fallback_creds,
+                               grpc_ce_channel_credentials) != SUCCESS) {
+    zend_throw_exception(spl_ce_InvalidArgumentException,
+                         "createXds expects a fallback credentials",
+                         1 TSRMLS_CC);
+    return;
+  }
+  char* fallback_creds_hash_str = "";
+  if (!fallback_creds) {
```

Comment:
I understand that we treat `NULL` as insecure credentials and that is not consistent with other languages. Because this involves security we want fallback to plaintext (insecure) to happen only when explicitly specified by the caller.

Reply:
`fallback_creds` is a required parameter being passed in from PHP userland. So the user would have to have done

```
$insecure_creds = ChannelCredentials::createInsecure();
createXds($insecure_creds);
```

for `fallback_creds` here to continue to be `NULL`. Would this work?

Reply:
So in the C-core land insecure_creds is translated to `NULL` I guess. What would be preferable (in my opinion) is to not special case insecure as NULL but rather `fallback_creds` to contain the right object even for insecure_creds so that after calling `PHP_GRPC_GET_WRAPPED_OBJECT(wrapped_grpc_channel_credentials, fallback_creds)` and accessing `wrapped_fallback_creds->wrapped` you would properly get the insecure_creds object so that the 'if-else` clause is unnecessary. But it's possible there is some quirk I am missing so ...

Reply:
I see. So you are saying that, let's have the PHP function `ChannelCredentials::createInsecure()` actually returns the object being returned by `grpc_insecure_credentials_create()`. Let's call this object X. Whether this X got treated as `NULL` in the c-core is irrelevant.  But if we do that, then when the user do

```
$insecure_creds = ChannelCredentials::createInsecure();   // contains X instead of NULL
createXds($insecure_creds);
```

by the time it reaches the code here, `wrapped_fallback_creds->wrapped` in L267 below would hopefully just contain X.

@HannahShiSFB Can you look into changing `ChannelCredentials::createInsecure()` to return `grpc_insecure_credentials_create()` instead of `NULL`?  This has implication on unit tests, generated code tests and interop tests. They all use `createInsecure()`. So please make sure those tests will still pass. Thanks!

Reply:
changed. unit tests and interop tests passed. one generated code tests case failed, the test case itself need to be fixed. 
@stanley-cheung I'll explain it offline.

Reply:
Does it mean this change uncovered an existing test-case issue? Just wondering...

Reply:
yes